rest/admin_api_test.go

```diff
@@ -33,6 +33,121 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// Test warnings being issued when a new channel is created with over 250 characters - CBG-1475
+func TestChannelNameSizeWarning(t *testing.T) {
+	syncFn := "function sync(doc, oldDoc) { channel(doc.chan); }"
+	rt := NewRestTester(t, &RestTesterConfig{
+		SyncFn: syncFn,
+	})
+	defer rt.Close()
+	getChannelNameWarnCount := func() int64 {
+		return rt.ServerContext().Database("db").DbStats.Database().WarnChannelNameSizeCount.Val
+	}
+
+	boundaryTestCases := []struct {
+		name               string
+		channelLengthToAdd int
+		expectWarn         bool
+	}{
+		{
+			name:               "Over max channel length",
+			channelLengthToAdd: 1,
+			expectWarn:         true,
+		},
+		{
+			name:               "Equal to max channel length",
+			channelLengthToAdd: 0,
+			expectWarn:         false,
+		},
+		{
+			name:               "Under max channel length",
+			channelLengthToAdd: -1,
+			expectWarn:         false,
+		},
+	}
+
+	boundaryTest := func(warnThresholdLength int) {
+		for _, test := range boundaryTestCases {
+			channelLength := warnThresholdLength + test.channelLengthToAdd
+			t.Run(fmt.Sprintf("%s: %d", test.name, channelLength), func(t *testing.T) {
+				chanNameWarnCountBefore := getChannelNameWarnCount()
+				tr := rt.SendAdminRequest("PUT", fmt.Sprintf("/db/doc%v", channelLength), fmt.Sprintf("{\"chan\":\"%s\"}", strings.Repeat("A", channelLength)))
+				assertStatus(t, tr, http.StatusCreated)
+				chanNameWarnCountAfter := getChannelNameWarnCount()
+				if test.expectWarn {
+					assert.Equal(t, chanNameWarnCountBefore+1, chanNameWarnCountAfter)
+				} else {
+					assert.Equal(t, chanNameWarnCountBefore, chanNameWarnCountAfter)
+				}
+			})
+		}
+	}
+
+	boundaryTest(int(base.DefaultWarnThresholdChannelNameSize))
+
+	// Update doc - should warn
+	channelLength := int(base.DefaultWarnThresholdChannelNameSize) + 5
+	t.Run("Update doc without changing channel", func(t *testing.T) {
```

Comment:
I think all these subtests make it difficult to read.

Are they independent of whats going on in the testCases loop?

If so:
- I'd prefer them as separate functions, or at least in another Test function that doesn't have the test case loop so its easier to immediately see each are standalone.

If not:
- Move them inside of the test case loop.